### PR TITLE
 feat: Add carpoolRequest module (action, reducer, saga) 

### LINKED ISSUE #24: [frontend] Add carpoolRequest module (action, reducer, saga)
### Subject of this issue

Need to add modules/carpoolRequest modules

### Detail Description

You need to write code about React action, reducer, redux-saga

This module should have following action type

* `REQUEST_CARPOOL`: On Rider side, the rider can request carpool through this action
* `CANCEL_CARPOOL`: On Rider side, the rider can cancel carpool through this action
* `MY_CARPOOL`: on Rider side, the rider can query my current carpool

### PR DESCRIPTION
**Pull request by mss8087**
### What is the issue this pull request related with?

resolved #24

### Your checklist for this pull request
🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [X] Code builds correctly
- [X] Created tests which fail without the change (if possible)
- [X] All tests passing
- [X] Extended the README / documentation, if necessary

### Description
Please describe your pull request.

💔Thank you!

### CODE REVIEW

**Comment by mss8087** [comment]
Oh, lib/api/carpool is not uploaded...
I don't know why 'git add .' is not working. I gonna check this.

File: frontend/src/lib/api/carpool/carpool.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import client from '../client';
+
+export function get(id) {
+  return client.get(`/api/v1/carpool/${id}`);
+}
+
+export function create({ user_id, from, to, minimum_passenger }) {
```
Comment (zeroFruit) [inline frontend/src/lib/api/carpool/carpool.js]: please change function parameter format into CamelCase

Comment (zeroFruit) [reply frontend/src/lib/api/carpool/carpool.js]: unify the function format into lambda (arrow function without return statement)

File: frontend/src/modules/carpoolRequest/carpoolRequest.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+] = createRequestActionTypes('carpoolRequest/CANCEL_CARPOOL');
+
+const [
+  MY_CARPOOL,
+  MY_CARPOOL_SUCCESS,
+  MY_CARPOOL_FAILURE,
+] = createRequestActionTypes('carpoolRequest/MY_CARPOOL');
+
+export const requestCarpool = createAction(REQUEST_CARPOOL);
+export const cancelCarpool = createAction(CANCEL_CARPOOL);
```
Comment (zeroFruit) [inline frontend/src/modules/carpoolRequest/carpoolRequest.js]: you should pass carpool id as action payload

```javascript
export const cancelCarpool = createAction(CANCEL_CARPOOL, carpoolId => carpoolId);
```

File: frontend/src/modules/carpoolRequest/carpoolRequest.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  CANCEL_CARPOOL_FAILURE,
+] = createRequestActionTypes('carpoolRequest/CANCEL_CARPOOL');
+
+const [
+  MY_CARPOOL,
+  MY_CARPOOL_SUCCESS,
+  MY_CARPOOL_FAILURE,
+] = createRequestActionTypes('carpoolRequest/MY_CARPOOL');
+
+export const requestCarpool = createAction(REQUEST_CARPOOL);
```
Comment (zeroFruit) [inline frontend/src/modules/carpoolRequest/carpoolRequest.js]: you should pass parameters as action payload
```javascript
export const requestCarpool = createAction(REQUEST_CARPOOL, ({ userId, from, ... }) => ({ userId, from, ... }));
```

File: frontend/src/modules/carpoolRequest/carpoolRequest.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+export const cancelCarpool = createAction(CANCEL_CARPOOL);
+export const myCarpool = createAction(MY_CARPOOL);
```
Comment (zeroFruit) [inline frontend/src/modules/carpoolRequest/carpoolRequest.js]: you should pass the carpool id as action payload

**Comment by zeroFruit** [comment]
@mss8087 please check test coverage!

**Comment by mss8087** [comment]
Please check coveage together on Wed.
I don't know why Travis CI makes lower coverage than before.

**Comment by zeroFruit** [comment]
@mss8087 
I've pulled your branches and tested. And the result was all test passed! Also, I've tracked Travis-ci, coveralls and the job was successfully finished. Somehow Travis-ci, coveralls hook does not work correctly.
So I'm going to approve and squash and merge forcefully.

**Pull request closed without merging**